base/special/log.jl

```diff
@@ -397,51 +397,167 @@ function log1p(x::Float32)
+(0x1.9c00000000000p-1, 0x1.bd087383be000p-3, -0x1.d4bc4595412b6p-45),
+(0x1.9a00000000000p-1, 0x1.c6ffbc6f01000p-3, -0x1.1ec72c5962bd2p-48),
+(0x1.9600000000000p-1, 0x1.db13db0d49000p-3, -0x1.aff2af715b035p-45),
+(0x1.9400000000000p-1, 0x1.e530effe71000p-3, 0x1.212276041f430p-51),
+(0x1.9200000000000p-1, 0x1.ef5ade4dd0000p-3, -0x1.a211565bb8e11p-51),
+(0x1.9000000000000p-1, 0x1.f991c6cb3b000p-3, 0x1.bcbecca0cdf30p-46),
+(0x1.8c00000000000p-1, 0x1.07138604d5800p-2, 0x1.89cdb16ed4e91p-48),
+(0x1.8a00000000000p-1, 0x1.0c42d67616000p-2, 0x1.7188b163ceae9p-45),
+(0x1.8800000000000p-1, 0x1.1178e8227e800p-2, -0x1.c210e63a5f01cp-45),
+(0x1.8600000000000p-1, 0x1.16b5ccbacf800p-2, 0x1.b9acdf7a51681p-45),
+(0x1.8400000000000p-1, 0x1.1bf99635a6800p-2, 0x1.ca6ed5147bdb7p-45),
+(0x1.8200000000000p-1, 0x1.214456d0eb800p-2, 0x1.a87deba46baeap-47),
+(0x1.7e00000000000p-1, 0x1.2bef07cdc9000p-2, 0x1.a9cfa4a5004f4p-45),
+(0x1.7c00000000000p-1, 0x1.314f1e1d36000p-2, -0x1.8e27ad3213cb8p-45),
+(0x1.7a00000000000p-1, 0x1.36b6776be1000p-2, 0x1.16ecdb0f177c8p-46),
+(0x1.7800000000000p-1, 0x1.3c25277333000p-2, 0x1.83b54b606bd5cp-46),
+(0x1.7600000000000p-1, 0x1.419b423d5e800p-2, 0x1.8e436ec90e09dp-47),
+(0x1.7400000000000p-1, 0x1.4718dc271c800p-2, -0x1.f27ce0967d675p-45),
+(0x1.7200000000000p-1, 0x1.4c9e09e173000p-2, -0x1.e20891b0ad8a4p-45),
+(0x1.7000000000000p-1, 0x1.522ae0738a000p-2, 0x1.ebe708164c759p-45),
+(0x1.6e00000000000p-1, 0x1.57bf753c8d000p-2, 0x1.fadedee5d40efp-46),
+(0x1.6c00000000000p-1, 0x1.5d5bddf596000p-2, -0x1.a0b2a08a465dcp-47))
 
 # Log implementation that returns 2 numbers which sum to give true value with about 68 bits of precision
-# Implimentation adapted from SLEEFPirates.jl
+# Since `log` only makes sense for positive exponents, we speed up the implimentation by stealing the sign bit
+# of the input for an extra bit of the exponent which is used to normalize subnormal inputs.
 # Does not normalize results.
-# Must be caused with positive finite arguments
-function _log_ext(d::Float64)
-    m, e = significand(d), exponent(d)
-    if m > 1.5
-        m *= 0.5
-        e += 1.0
-    end
-    # x = (m-1)/(m+1)
-    mp1hi = m + 1.0
-    mp1lo = m + (1.0 - mp1hi)
-    invy = inv(mp1hi)
-    xhi = (m - 1.0) * invy
-    xlo = fma(-xhi, mp1lo, fma(-xhi, mp1hi, m - 1.0)) * invy
-    x2hi, x2lo = two_mul(xhi, xhi)
-    x2lo = muladd(xhi, xlo * 2.0, x2lo)
-    thi, tlo  = log_ext_kernel(x2hi, x2lo)
-
-    shi = 0.6931471805582987 * e
-    xhi2 = xhi * 2.0
-    shinew = muladd(xhi, 2.0, shi)
-    slo = muladd(1.6465949582897082e-12, e, muladd(xlo, 2.0, (((shi - shinew) + xhi2))))
-    shi = shinew
-    x3hi, x3lo = two_mul(x2hi, xhi)
-    x3lo = muladd(x2hi, xlo, muladd(xhi, x2lo,x3lo))
-    x3thi, x3tlo = two_mul(x3hi, thi)
-    x3tlo = muladd(x3hi, tlo, muladd(x3lo, thi, x3tlo))
-    anshi = x3thi + shi
-    anslo = slo + x3tlo - ((anshi - shi) - x3thi)
-    return anshi, anslo
+# Copyright (c) 2018-2020, Arm Limited.
```

Comment:
would be good to add the link from the PR body here

Reply:
Note also that the ARM code is dual licensed MIT and Apache 2.0 + exceptions, so it'd be good to specify which of the licenses the code was being used under for reference purposes.

Reply:
(This also isn't the software itself nor a substantial portion of it, it's a separate implementation based on a reference, so I think that per the terms of the MIT license, acknowledging the original license is more documentation/good form/courtesy than a legal requirement AFAIU, though IANAL.)